contracts/command-batcher/src/msg.rs

```diff
@@ -1,13 +1,30 @@
+use std::collections::HashMap;
+
 use cosmwasm_schema::{cw_serde, QueryResponses};
-use cosmwasm_std::HexBinary;
+use cosmwasm_std::{HexBinary, Uint256, Uint64};
+
+use crate::types::{Data, Proof, ProofID};
 
-use crate::types::{Data, Proof};
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub gateway_address: String,
+    pub multisig_address: String,
+    pub service_registry_address: String,
+    pub destination_chain_id: Uint256,
+    pub signing_threshold: (Uint64, Uint64),
+    pub service_name: String,
+}
 
 #[cw_serde]
 pub enum ExecuteMsg {
     // Start building a proof that includes specified messages
     // Queries the gateway for actual message contents
-    ConstructProof { message_ids: Vec<String> },
+    ConstructProof {
+        message_ids: Vec<String>,
+    },
+    KeyGen {
+        pub_keys: HashMap<String, HexBinary>,
+    },
```

Comment:
Can we name this `RotateSnapshot`, to be more compatible with the changes we have planned?